### PR TITLE
Code coverage test

### PR DESCRIPTION
**Pull request by eNeRGy164**
## IMPORTANT 

* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

**Comment by dennisdoomen** [comment]
I'd love to add that to the repo, but then through a target in Nuke. I want to keep that YAML to a minimum at all cost. Advantage of that is that you get the coverage even if you run the build locally.

**Comment by eNeRGy164** [comment]
I've kept it to only using coveralls for the reporting, it keeps the build config much more simple for now, only having coveralls upload in it.
I think coveralls is useful for having a trend over time. This is is possible with report generator, but doesn't make much sense on a local system.

I moved report generator (which was needed for combining output anyway) to Nuke, so there will be a local report, I report the URL in the build output for convenience.

**Comment by dennisdoomen** [comment]
Is it supposed to render the results in the PR somewhere?

**Comment by eNeRGy164** [comment]
That will only happen if there is a baseline against the master branch.

**Comment by dennisdoomen** [comment]
Would you mind pushing this as an official PR?

File: Build/Build.cs [hunk 1]
```diff
@@ -132,9 +135,20 @@ class Build : NukeBuild
...
                 .SetConfiguration("Debug")
                 .EnableNoBuild()
+                .SetDataCollector("XPlat Code Coverage")
+                .SetResultsDirectory(RootDirectory / "TestResults")
                 .CombineWith(
                     Solution.Specs.FluentAssertions_Specs.GetTargetFrameworks().Except(new[] { "net47" }),
                     (_, v) => _.SetFramework(v)));
+
+            ReportGenerator(s => s
+                .SetProcessToolPath(ToolPathResolver.GetPackageExecutable("ReportGenerator", "ReportGenerator.exe", framework: "net5.0"))
```
Comment (jnyrup) [inline Build/Build.cs:145]: Does the `ReportGenerator` tool support specifying the path to `ReportGenerator.dll` instead?
That should make it less windows specific.

File: Build/Build.cs [hunk 2]
```diff
@@ -132,9 +135,20 @@ class Build : NukeBuild
...
+                .SetTargetDirectory(RootDirectory / "TestResults" / "reports")
+                .AddReports(RootDirectory / "TestResults/**/coverage.cobertura.xml")
+                .AddReportTypes("HtmlInline_AzurePipelines_Dark", "lcov")
+                .SetAssemblyFilters("-AssemblyA;+*"));
```
Comment (jnyrup) [inline Build/Build.cs:149]: Would it be easier just to specify `"FluentAssertions"` here?

File: Build/Build.cs (hunk 1 above)
Comment (dennisdoomen) [reply Build/Build.cs]: You guys realize that this PR is targeting Michaël's fork and not the main repo?

**Pull request merged**